pkg/cluster/kubeclient/backup.go

```diff
@@ -0,0 +1,84 @@
+package kubeclient
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/openshift-azure/pkg/util/ready"
+)
+
+func (u *kubeclient) BackupCluster(ctx context.Context, backupImage, backupName string) error {
+	u.log.Infof("running an etcd backup using %s", backupImage)
+	namespace := "openshift-etcd"
+	bk, err := u.client.BatchV1().Jobs(namespace).Create(&batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      backupName,
+			Namespace: namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					NodeSelector:       map[string]string{"node-role.kubernetes.io/master": "true"},
+					ServiceAccountName: "etcd-backup",
+					RestartPolicy:      "Never",
+					Containers: []v1.Container{
+						{
+							Name:            backupName,
+							Image:           backupImage,
+							ImagePullPolicy: "Always",
+							Args:            []string{fmt.Sprintf("-blobname=%s", backupName), "save"},
```

Comment:
You need to validate backupName before this step.  As you are also using it for the object name, check that it is a valid object name, and that should be safe as an arg too.

Reply:
i'm not going to use the backupName for the job/container name. I'll use the existing `etcd-backup` name